Neos.ContentRepository.Core/Classes/Subscription/Engine/SubscriptionEngine.php

```diff
@@ -240,101 +253,112 @@ private function catchUpSubscriptions(SubscriptionEngineCriteria $criteria, Subs
     {
         $this->logger?->info(sprintf('Subscription Engine: Start catching up subscriptions in state "%s".', $subscriptionStatus->value));
 
-        return $this->subscriptionManager->findForAndUpdate(
+        $subscriberGroups = $this->subscribers->groupByStore();
+
+        // todo merge results
+        $returnResult = null;
+
+        foreach ($subscriberGroups as [$store, $subscribers]) {
+            // todo do not global override manager!
+            $this->subscriptionManager = new SubscriptionManager($store);
+            $returnResult = $this->subscriptionManager->findForAndUpdate(
             SubscriptionCriteria::forEngineCriteriaAndStatus($criteria, $subscriptionStatus),
-            function (Subscriptions $subscriptions) use ($subscriptionStatus, $progressClosure) {
-                foreach ($subscriptions as $subscription) {
-                    if (!$this->subscribers->contain($subscription->id)) {
-                        // mark detached subscriptions as we cannot handle them and exclude them from catchup
-                        $subscription->set(
-                            status: SubscriptionStatus::DETACHED,
-                        );
-                        $this->subscriptionManager->update($subscription);
-                        $this->logger?->info(sprintf('Subscription Engine: Subscriber for "%s" not found and has been marked as detached.', $subscription->id->value));
-                        $subscriptions = $subscriptions->without($subscription->id);
+                function (Subscriptions $subscriptions) use ($subscriptionStatus, $progressClosure, $store, $subscribers) {
+                    foreach ($subscriptions as $subscription) {
+                        // TODO we cannot mark a subscriber as detached, if it belongs to another store and no one else is using that store anymore. Then it will just be ACTIVE and never found, not even by status
+                        if (!$subscribers->contain($subscription->id)) {
```

Comment:
but if the  package is reinstalled, the old database entry will be found stating that it is still ACTIVE meaning that any interaction leading to a catchup active will lead to that the possibly really far behind projection will get a full update